dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/error/DefaultRuntimeErrorCallback.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dashbuilder.client.error;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import com.google.gwt.user.client.Window;
+import org.dashbuilder.client.resources.i18n.AppConstants;
+import org.jboss.errai.bus.client.api.InvalidBusContentException;
+import org.uberfire.ext.widgets.common.client.common.BusyIndicatorView;
+import org.uberfire.ext.widgets.common.client.common.popups.YesNoCancelPopup;
+import org.uberfire.ext.widgets.common.client.resources.i18n.CommonConstants;
+
+import static org.uberfire.ext.widgets.common.client.common.popups.YesNoCancelPopup.newYesNoCancelPopup;
+import static org.uberfire.ext.widgets.common.client.common.popups.errors.ErrorPopup.showMessage;
+
+@Dependent
+public class DefaultRuntimeErrorCallback {
+
+    @Inject
+    BusyIndicatorView loading;
+
+    AppConstants i18n = AppConstants.INSTANCE;
+
+    private boolean errorPopUpLock = false;
+
+    public void error(final Throwable throwable) {
+        if (errorPopUpLock) {
+            return;
+        }
+        loading.hideBusyIndicator();
+        errorPopUpLock = true;
+        if (isServerOfflineException(throwable)) {
+            yesNoPopup(i18n.DisconnectedFromServer(), i18n.CouldNotConnectToServer());
+        } else if (isInvalidBusContentException(throwable)) {
+            yesNoPopup(i18n.SessionTimeout(), i18n.InvalidBusResponseProbablySessionTimeout());
+        } else {
+            showMessage(CommonConstants.INSTANCE.ExceptionGeneric0(extractMessageRecursively(throwable)),
+                        this::unlock,
+                        this::unlock);
+        }
+
+    }
+
+    private void yesNoPopup(String title, String content) {
+        final YesNoCancelPopup result = newYesNoCancelPopup(title, content, Window.Location::reload, this::unlock, this::unlock);
```

Comment:
Hello @caponetto - thanks for great suggestion. The issue is that there's no way to change the popUp content after it is created, but I will change the popUp body widget and see it works.

Reply:
Oh, in this case, let keep the way you did then.
I saw a `setTitle` method and thought there could be a `setContent` as well, which is not the case.

Reply:
I tried your suggestion, but I had to: keep the flag since isVisible was not working, then setting the Body widget didn't work, it messed the popup, so I will have to recreate it everytime :( 

![image](https://user-images.githubusercontent.com/359820/89798182-4018b500-db02-11ea-8d18-3c3c94760b6f.png)

Reply:
Thanks for trying, @jesuino! We've learned something today :)